rust/datafusion/src/execution/physical_plan/expressions.rs

```diff
@@ -196,6 +197,61 @@ pub fn sum(expr: Arc<dyn PhysicalExpr>) -> Arc<dyn AggregateExpr> {
     Arc::new(Sum::new(expr))
 }
 
+/// CAST expression casts an expression to a specific data type
+pub struct CastExpr {
+    /// The expression to cast
+    expr: Arc<dyn PhysicalExpr>,
+    /// The data type to cast to
+    cast_type: DataType,
+}
+
+/// Determine if a DataType is numeric or not
+fn is_numeric(dt: &DataType) -> bool {
+    match dt {
+        DataType::Int8 | DataType::Int16 | DataType::Int32 | DataType::Int64 => true,
+        DataType::UInt8 | DataType::UInt16 | DataType::UInt32 | DataType::UInt64 => true,
+        DataType::Float16 | DataType::Float32 | DataType::Float64 => true,
+        _ => false,
+    }
+}
+
+impl CastExpr {
+    /// Create a CAST expression
+    pub fn try_new(
+        expr: Arc<dyn PhysicalExpr>,
+        input_schema: &Schema,
+        cast_type: DataType,
+    ) -> Result<Self> {
+        let expr_type = expr.data_type(input_schema)?;
+        // numbers can be cast to numbers and strings
+        if is_numeric(&expr_type)
+            && (is_numeric(&cast_type) || cast_type == DataType::Utf8)
+        {
+            Ok(Self { expr, cast_type })
+        } else {
+            Err(ExecutionError::General(format!(
+                "Invalid CAST from {:?} to {:?}",
+                expr_type, cast_type
+            )))
+        }
+    }
+}
+
+impl PhysicalExpr for CastExpr {
+    fn name(&self) -> String {
+        "CAST".to_string()
```

Comment:
nit: there is inconsistency in upper vs lower case for `name`.  The literal one was lower case, this one and SUM were upper case.